catalogs/catalog-model/src/main/java/org/apache/gravitino/catalog/model/ModelCatalogOperations.java

```
@@ -614,6 +625,18 @@ private void doRemoveProperty(
     entityProperties.remove(change.property());
   }
 
+  private void doUpdateUri(Map<String, String> entityUris, ModelVersionChange.UpdateUri change) {
+    entityUris.replace(change.uriName(), change.newUri());
+  }
+
+  private void doAddUri(Map<String, String> entityUris, ModelVersionChange.AddUri change) {
+    entityUris.putIfAbsent(change.uriName(), change.uri());
+  }
+
+  private void doRemoveUri(Map<String, String> entityUris, ModelVersionChange.RemoveUri change) {
+    entityUris.remove(change.uriName());
+  }
```

What are the behaviors if:

1. The update URI name doesn't exist.
2. The added URI name already exists.
3. The removed URI doesn't exist.

Shall we throw an exception, or do it silently, what do you think?

---

Maybe we'd better throw the related exception if "replace" has no existing uri name, or "put" has existing name, and "delete" has no existing name.

---

I checked similar logic, like removing the property of an entity, and none of them threw an exception in this case, maybe we should keep consistent?

---

OK, I see.